### DISCUSSION TITLE
It seems impossible to associate warnings with the VIPS operation causing them when using multithreading

### DISCUSSION THREAD

**Discussion by mika-fischer** [libvips/libvips#3026] [General]
We're using VIPS to load images from multiple threads. Since (at least some) warnings seem to be called from the VIPS threadpool, I can't figure out a way to associate a given warning with the VIPS operation that causes the warning, or the thread that started the VIPS operation.

Am I missing something here? Or is this really a limitation of the VIPS API?

**Comment by jcupitt** [comment]
Hi @mika-fischer, 

That's right, libvips has a single log for error messages for the whole process. You get success/fail on the API, but no more detail than that.

It's OK for a single threaded application (in my opinion), partly because libvips is lazy and there's not an exact match between API calls and error events. But I agree that a better system is probably needed for large, multithreaded server applications where several distinct tasks are being worked on at once.
